examples/services/extension_example.py

```diff
@@ -0,0 +1,41 @@
+import asyncio
+import logging.config
+from pathlib import Path
+
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.core.config.loader import BdkConfigLoader
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+from symphony.bdk.core.symphony_bdk import SymphonyBdk
+
+
+class MyExtension:
+    def __init__(self):
+        self._bot_session = None
+        self._config = None
+
+    def set_auth_session(self, auth_session: AuthSession):
+        logger.debug("Registering auth session")
+        self._bot_session = auth_session
+
+    def set_configuration(self, bdk_config: BdkConfig):
+        logger.debug("Registering config: " + bdk_config.host)
+        self._config = bdk_config
+
+    def get_service(self):
+        return self
+
+    async def get_session_token(self):
+        return await self._bot_session.session_token
+
+
+async def run():
+    async with SymphonyBdk(BdkConfigLoader.load_from_symphony_dir("config.yaml")) as bdk:
+        bdk.extensions().register(MyExtension)
+        service = bdk.extensions().get_service(MyExtension)
```

Comment:
you might even wonder, do we need to register and get the service? vs just getting the service

Reply:
Yes, I remember we discussed that during the design of the extension API in the Java BDK.
* We could have extension that don't want to expose services (like background tasks)
* I guess we want to stay close to the Java BDK API to not confuse customers

Reply:
An open question: do we want to give the ability to register extensions by specifying a package name (as a string)? Do we want to give the ability to get a service by specifying a string, like `"my.package.MyExtension"`? 
If we go for that, implementation will be slightly different as we will have to dynamically import the extensions using importlib.

Reply:
given you have the extension in your project at that stage, is there a value in doing it dynamically?